test/e2e/setup/setup.go

```diff
@@ -46,7 +46,7 @@ func Setup(t *testing.T) *e2eutil.Context {
 		t.Fatal(err)
 	}
 
-	if err := deployOperator(); err != nil {
+	if err := deployOperator(ctx); err != nil {
 		t.Fatal(err)
 	}
 
```

Comment:
I had to pass ctx because before createOrUpdate was called with an empty context which meant that the shouldPerformCleanup flag was not used.